src/mlx/warnings.py

```diff
@@ -13,45 +14,99 @@
 
 
 class WarningsPlugin:
-    def __init__(self):
-        self.sphinx_counter = 0
-        self.doxygen_counter = 0
-        self.junit_counter = 0
 
-    def check_sphinx_warnings(self, line):
+    def __init__(self, sphinx = False, doxygen = False, junit = False):
```

Comment:
- Why not use the lookup table approach instead? This will throw an expection when the string cannot be found in the dictionary, which is more pythonic than return codes IMO
- If this doesn't seem right to you, maybe you could define a sort of configuration structure that could be passed on to the constructor (or to the ```activate_checker``` function)? This would also allow adding min and max values per checker.